src/cashlib/cashlib.cpp

```diff
@@ -154,7 +157,7 @@ extern "C" int SignTx(unsigned char *txData,
     CKey key = LoadKey(keyData);
 
     size_t nHashedOut = 0;
-    uint256 sighash = SignatureHashBitcoinCash(priorScript, tx, inputIdx, nHashType, inputAmount, &nHashedOut);
```

Comment:
why not use SignatureHashBitcoinCash here?

Reply:
Well, my general guideline was to always try to reduce the surface area and thus coupling between the different parts of `bitcoind`. What would be the reason to be more specific here?

Reply:
this change increases the surface area.  SignatureHash calls either ...Legacy or ...BitcoinCash depending on flags.  I wanted to restrict cashlib to BitcoinCash signature hash only to have one less thing that can go wrong for the wallet dev (accidentally Legacy signing in the BitcoinCash network)...